test/object-store/sync/flx_sync.cpp

```diff
@@ -255,6 +266,96 @@ TEST_CASE("flx: query on non-queryable field results in query error message", "[
     });
 }
 
+TEST_CASE("flx: interrupted bootstrap restarts/recovers on reconnect", "[sync][flx][app]") {
+    FLXSyncTestHarness harness("flx_bootstrap_batching", {g_large_array_schema, {"queryable_int_field"}});
+
+    // First we need to seed the server with objects that are large and complex enough that they get broken
+    // into multiple download messages.
+    //
+    // The server will break up changesets and download messages when they contain more than 1000 instructions
+    // and are bigger than 1MB respectively.
+    //
+    // So this generates 5 objects each with 1000+ instructions that are each 1MB+ big. This should result in
+    // 3 download messages total with one changeset each for the bootstrap download messages.
+    std::vector<ObjectId> obj_ids_at_end;
+    harness.load_initial_data([&](SharedRealm realm) {
+        CppContext c(realm);
+        for (int i = 0; i < 5; ++i) {
+            auto id = ObjectId::gen();
+            auto obj = Object::create(c, realm, "TopLevel",
+                                      util::Any(AnyDict{{"_id", id},
+                                                        {"list_of_strings", AnyVector{}},
+                                                        {"queryable_int_field", static_cast<int64_t>(i * 5)}}));
+            List str_list(obj, realm->schema().find("TopLevel")->property_for_name("list_of_strings"));
+            for (int j = 0; j < 1024; ++j) {
+                str_list.add(c, util::Any(std::string(1024, 'a' + (j % 26))));
+            }
+
+            obj_ids_at_end.push_back(id);
+        }
+    });
+    SyncTestFile interrupted_realm_config(harness.app()->current_user(), harness.schema(),
+                                          SyncConfig::FLXSyncEnabled{});
+    interrupted_realm_config.cache = false;
+
+    {
+        SharedRealm realm;
+        auto [interrupted_promise, interrupted] = util::make_promise_future<void>();
+        Realm::Config config = interrupted_realm_config;
+        config.sync_config->on_download_message_received_hook =
+            [download_msg_counter = int(0),
+             promise = std::make_shared<util::Promise<void>>(std::move(interrupted_promise))](
+                std::weak_ptr<SyncSession> weak_session) mutable {
+                auto session = weak_session.lock();
+                // We interrupt on the 5rd download message, which should be 1/3rd of the way through the
```

Comment:
this comment seems wrong. aren't 5 download messages in total? If that's the case, then bootstrapping should be completed, but that contradicts with the check at line 336. I guess the whole idea is to re-open the same realm afterwards. Is that the case?